src/data/finset/basic.lean

```diff
@@ -2238,6 +2238,42 @@ finset.strong_induction_on s $ λ s,
 finset.induction_on s (λ _, h₀) $ λ a s n _ ih, h₁ a s n $
 λ t ss, ih _ (lt_of_le_of_lt ss (ssubset_insert n) : t < _)
 
+/-- Suppose that, given `s.card ≤ n` and objects defined on all strict supersets of card less than
+`n` of any finset, one knows how to define an object on `s`. Then one can inductively define an
+object on all finsets of card less than `n`, starting from finsets of card `n` and iterating. This
+can be used either to define data, or to prove properties. -/
+def strong_downward_induction {p : finset α → Sort*} {n : ℕ} (H : ∀ t₁, (∀ {t₂ : finset α},
+  t₂.card ≤ n → t₁ ⊂ t₂ → p t₂) → t₁.card ≤ n → p t₁) :
+  ∀ (s : finset α), s.card ≤ n → p s
+| s := H s (λ t ht h, have n - card t < n - card s, begin
+  exact (nat.sub_lt_sub_left_iff ht).2 (finset.card_lt_card h),
+end, strong_downward_induction t ht)
```

Comment:
No need for `begin` `end` here, since the proof is just `exact`.
```suggestion
| s := H s (λ t ht h, 
  have n - card t < n - card s, 
     from (nat.sub_lt_sub_left_iff ht).2 (finset.card_lt_card h), 
  strong_downward_induction t ht)
```